### PR TITLE
Update substrait to 0.7.0

### PR DESCRIPTION
**Pull request by PenghuiJiao**

### CODE REVIEW

**Comment by PenghuiJiao** [comment]
please take review https://github.com/orgs/intel-innersource/teams/intel-modular-sql-plan-team

**Comment by mbasmanova** [comment]
@PenghuiJiao Thank you for updating Substrait. By convention, PRs that update dependencies should use the following format for the PR title: "Update Xxx to <new version> (from <old version>)". Would you update accordingly?

File: velox/substrait/SubstraitToVeloxPlan.cpp [hunk 1]
```diff
@@ -340,11 +340,11 @@ core::PlanNodePtr SubstraitVeloxPlanConverter::toVeloxPlan(
       splitInfo->paths.emplace_back(file.uri_file());
       splitInfo->starts.emplace_back(file.start());
       splitInfo->lengths.emplace_back(file.length());
-      switch (file.format()) {
-        case 0:
+      switch (file.file_format_case()) {
+        case 11:
```
Comment (mbasmanova) [inline velox/substrait/SubstraitToVeloxPlan.cpp:344]: Does Substrait provide constants for the values? Using raw numbers makes the code hard to read.

**Review by mbasmanova — APPROVED** [review]
@PenghuiJiao Looks good. Would you rebase onto latest master to pick up recent change to CircleCI configuration which adds a job to run Substrait tests.

**Comment by PenghuiJiao** [comment]
> @PenghuiJiao Thank you for updating Substrait. By convention, PRs that update dependencies should use the following format for the PR title: "Update Xxx to (from )". Would you update accordingly?

I have modified the title

**Comment by mbasmanova** [comment]
@PenghuiJiao Thank you for updating the title. It reads "Update substrait to 0.7.0", but is missing the "(from yyy)". What was the version used before the upgrade?

File: velox/substrait/SubstraitToVeloxPlan.cpp (hunk 1 above)
Comment (PenghuiJiao) [reply velox/substrait/SubstraitToVeloxPlan.cpp]: I have used enums instead of numbers

File: velox/substrait/SubstraitToVeloxPlan.cpp [hunk 2]
```diff
@@ -341,10 +342,10 @@ core::PlanNodePtr SubstraitVeloxPlanConverter::toVeloxPlan(
       splitInfo->starts.emplace_back(file.start());
       splitInfo->lengths.emplace_back(file.length());
       switch (file.file_format_case()) {
-        case 11:
+        case SubstraitFileFormatCase::kOrc:
```
Comment (mbasmanova) [inline velox/substrait/SubstraitToVeloxPlan.cpp:345]: Nice. So much easier to understand.

**Comment by mbasmanova** [comment]
> Update substrait to v0.7.0 from substrait-io

"from xxx" should specify the version that was used before the upgrade.

File: velox/substrait/SubstraitToVeloxPlan.cpp (hunk 1 above)
Comment (PenghuiJiao) [reply velox/substrait/SubstraitToVeloxPlan.cpp]: I have rebased onto latest master

Comment (mbasmanova) [reply velox/substrait/SubstraitToVeloxPlan.cpp]: Thanks. format-check job is failing. Would you take a look?

Comment (PenghuiJiao) [reply velox/substrait/SubstraitToVeloxPlan.cpp]: Thanks，I have formatted the code

Comment (PenghuiJiao) [reply velox/substrait/SubstraitToVeloxPlan.cpp]: but, macos-build fail and test fail.

Comment (mbasmanova) [reply velox/substrait/SubstraitToVeloxPlan.cpp]: Hmm... Make sure you are rebased onto lates main.

File: velox/substrait/CMakeLists.txt [hunk 3]
```diff
@@ -33,7 +33,7 @@ get_filename_component(PROTO_DIR ${substrait_proto_directory}/, DIRECTORY)
 # Generate Substrait hearders
 add_custom_command(
   OUTPUT ${PROTO_OUTPUT_FILES}
-  COMMAND protoc --proto_path ${CMAKE_SOURCE_DIR}/ --cpp_out ${CMAKE_SOURCE_DIR}
+  COMMAND protoc --proto_path ${proto_directory}/ --cpp_out ${PROTO_OUTPUT_DIR}
```
Comment (mbasmanova) [inline velox/substrait/CMakeLists.txt:36]: Why is this change? Can it be responsible for macos build failure?

Comment (PenghuiJiao) [reply velox/substrait/CMakeLists.txt:36]: Krishna Pai change it at commit:52d1dcc14e2a69da2e244b86bfe00edab078aac6  on July 14

Comment (mbasmanova) [reply velox/substrait/CMakeLists.txt:36]: Actually, looks like this PR reverts Krishna's fix. Remove this change and retry.

Comment (PenghuiJiao) [reply velox/substrait/CMakeLists.txt:36]: use   COMMAND protoc --proto_path ${CMAKE_SOURCE_DIR}/ --cpp_out ${CMAKE_SOURCE_DIR} , need  import "velox/substrait/proto/substrait/xxx.proto";

use  COMMAND protoc --proto_path ${proto_directory}/ --cpp_out ${PROTO_OUTPUT_DIR}， need import "substrait/xxx.proto";
But the source code of substrait-io is import "substrait/xxx.proto";

Comment (PenghuiJiao) [reply velox/substrait/CMakeLists.txt]: I had remove this change and retry.

**Comment by mbasmanova** [comment]
@PenghuiJiao Would you try rebasing onto latest once again?

**Comment by mbasmanova** [comment]
@PenghuiJiao 

I'm seeing test failures: Substrait2VeloxPlanConversionTest.q6 and FunctionTest.constructFunctionMap. Are these passing for you?

File: velox/substrait/CMakeLists.txt (hunk 3 above)
Comment (kgpai) [reply velox/substrait/CMakeLists.txt]: @PenghuiJiao Is this working now ? fyi protoc --proto_path ${CMAKE_SOURCE_DIR}/ --cpp_out ${CMAKE_SOURCE_DIR} should work for all.

File: velox/substrait/SubstraitToVeloxPlan.cpp (hunk 1 above)
Comment (kgpai) [reply velox/substrait/SubstraitToVeloxPlan.cpp]: Again there should be no need to make changes to the CMakelists file. Please reach out to me on slack @kgpai and maybe i can help you uncover why the brew install is failing for you @PenghuiJiao .

**Comment by mbasmanova** [comment]
I noticed that Substrait tests do not run in CircleCI. @kgpai Krishna is looking into how to add them so we can get a clear signal.

File: velox/substrait/CMakeLists.txt (hunk 3 above)
Comment (kgpai) [reply velox/substrait/CMakeLists.txt]: nit: if you dont quote github turns the `${CMAKE_SOURCE_DIR}` to a wierd font.

Comment (PenghuiJiao) [reply velox/substrait/CMakeLists.txt]: > @PenghuiJiao
> 
> I'm seeing test failures: Substrait2VeloxPlanConversionTest.q6 and FunctionTest.constructFunctionMap. Are these passing for you?

These tests failed

**Comment by PenghuiJiao** [comment]
> @PenghuiJiao
> 
> I'm seeing test failures: Substrait2VeloxPlanConversionTest.q6 and FunctionTest.constructFunctionMap. Are these passing for you?

These tests failed

File: velox/substrait/SubstraitToVeloxPlan.cpp (hunk 1 above)
Comment (PenghuiJiao) [reply velox/substrait/SubstraitToVeloxPlan.cpp]: Can you invite me to join your group on slack? @kgpai

Comment (kgpai) [reply velox/substrait/SubstraitToVeloxPlan.cpp]: I couldnt add you myself . I have requested @jijufb to add you to Velox on slack.

Comment (jijufb) [reply velox/substrait/SubstraitToVeloxPlan.cpp]: Invite is sent on Velox Slack Channel.

**Comment by zhejiangxiaomai** [comment]
> _ No description provided. _

Sub.json is replaced with q6_first_stage.json. because this json file represents the query plan of TPCH-6.

File: velox/substrait/tests/Substrait2VeloxPlanConversionTest.cpp [hunk 4]
```diff
@@ -268,9 +268,9 @@ TEST_F(Substrait2VeloxPlanConversionTest, q6) {
 
   // Find and deserialize Substrait plan json file.
   std::string planPath =
-      getDataFilePath("velox/substrait/tests", "data/sub.json");
```
Comment (mbasmanova) [inline velox/substrait/tests/Substrait2VeloxPlanConversionTest.cpp:271]: @PenghuiJiao Is sub.json not used anymore? If so, let's delete it.

Comment (mbasmanova) [reply velox/substrait/tests/Substrait2VeloxPlanConversionTest.cpp:271]: Got it. Thank you for clarifying. I'm going to re-run tests and land if they pass. Next time, please, remember to update PR description when making additional changes.

**Pull request closed without merging**